### DISCUSSION TITLE
"wire:navigate" previous page 404 error

### DISCUSSION THREAD

**Discussion by melikceran** [livewire/livewire#6095] [7. Bugs (archived)]
### Livewire version

v3.0.0-beta.5

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/melikceran/livewire-404-demo

### Steps To Reproduce

I have a Livewire component called "Navigation". I use the "wire:navigate" directive for the links in the "Navigation" component, and the "Navigation" component is rendered again when there are any changes to these links.
However, the problem is that; If the laravel route address of any of the links I visit with "wire:navigate" is removed and rendered again, I get a 404 error as if I was on that page, even though I am not on that page.

1) I visit /example-page and then /other-page.
2) I am removing laravel Route access for /example-page.
3) "Navigation" Component is being rerendered.
4) I am at /other-page but it gives 404 error as if I am at /example-page.

So if the previously visited page URLs have been removed, the component gives a 404 error when rendered.

POST /livewire/update
**payload:**
`{
    "_token": "xoeNSJkyDdGVHjWJ3TMZ6xbVzWHJ9G5qi8Fo9849",
    "components": [
        {
            "snapshot": "{\"data\":[],\"memo\":{\"id\":\"zhDiWr4qeIHlh2bSxWMv\",\"name\":\"navigation\",\"path\":\"example-page\",\"method\":\"GET\",\"children\":[],\"lazyLoaded\":false,\"errors\":[],\"locale\":\"en\"},\"checksum\":\"cb23e3519a1c8f61217ed2b5ec52fe7101a08db1888a690e306d5c5ed8313963\"}",
            "updates": {},
            "calls": [
                {
                    "path": "",
                    "method": "__dispatch",
                    "params": [
                        "refresh",
                        {}
                    ]
                }
            ]
        },
        {
            "snapshot": "{\"data\":[],\"memo\":{\"id\":\"p4cgddyRru7nIWQobNXS\",\"name\":\"navigation\",\"path\":\"other-page\",\"method\":\"GET\",\"children\":[],\"lazyLoaded\":false,\"errors\":[],\"locale\":\"en\"},\"checksum\":\"974c7bca525e5b3badf3e2aba2cde67a60e164081ab0bedc144ea8f97b0daa45\"}",
            "updates": {},
            "calls": [
                {
                    "path": "",
                    "method": "__dispatch",
                    "params": [
                        "refresh",
                        {}
                    ]
                }
            ]
        }
    ]
}`

Request URL: http://livewire.test/livewire/update
Request Method: POST
Status Code: 404 Not Found

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by romansh** [comment]
Can you provide reproducible pieces of code? Like a component with its view?

**Reply by melikceran** [reply]
I have attached the files for your review.

**Comment by PhiloNL** [comment]
If you can provide a demo or failing test, that would be great.

**Reply by melikceran** [reply]
I have attached the files for your review.

**Comment by melikceran** [comment]
I created an demo. To catch the error, the Navigation component refreshes itself every 2 seconds.

1) Click on all pages named Welcome Page, Example Page, Other Page.
2) Then disable a route url you visited in the past in route/web.php.
3) When the navigation component refreshes itself within 2 seconds, it will give a 404 error even if you are not on the page you disabled.

https://github.com/melikceran/livewire-404-demo

[livewire-demo.zip](https://github.com/livewire/livewire/files/12210078/livewire-404.zip)

When the component is refreshed, it sends the list of previously visited pages. If one of the pages does not exist, it will throw an error.
![livewire-404](https://github.com/livewire/livewire/assets/65822163/ee7ecaf2-a92a-44d4-a802-cf9e61931c7a)

I examined the livewire.js codes related to the error.
I found the error in the event listener, although I don't fully understand its functions. As you can see in the picture, it doesn't give an error when I disable the listener. However, since I don't know where else the code is used, I wanted to provide information to help.

![livewire-404](https://github.com/livewire/livewire/assets/65822163/fd89da0d-0bbd-408f-95e3-d747474b6422)

**Reply by joshhanley** [reply]
@melikceran thanks for the demo, but we won't open `.zip` files, can you please upload it to a public repo on GitHub and share the link? Thanks

**Reply by melikceran** [reply]
https://github.com/melikceran/livewire-404-demo

**Comment by melikceran** [comment]
https://github.com/livewire/livewire/assets/65822163/3b0e0330-e23c-4d08-9160-37f5a8f436b7

Demo: https://github.com/melikceran/livewire-404-demo

**Reply by joshhanley** [reply]
Looking at the payload, it seems that Livewire is still trying to dispatch an event to the "Example Page" component even though it's no longer on the page. See screenshot below. I will see if Caleb can have a look.

<img width="1529" alt="image" src="https://github.com/livewire/livewire/assets/882837/e8f3fa1c-115a-4e9e-86f1-eb6bc0fede30">

**Reply by melikceran** [reply]
I think it should be on the priority list as an important bug. Because wire:navigate does not comply with the usage logic. I have to reload the Page to avoid getting the error.

**Reply by joshhanley** [reply]
@melikceran currently we're in beta, so there is no priority list as such. Each bug is important, but you shouldn't be using this for anything critical until we release a stable version of V3 🙂

**Reply by riasadrion** [reply]
I am on livewire 3, still events are duplicated after wire:navigate. It is a nice feature but can't handle events
